unittests/internaltests.py

```diff
@@ -1648,7 +1648,7 @@ def test_interpreter_unpicklable(self) -> None:
         build = mock.Mock()
         build.environment = mock.Mock()
         build.environment.get_source_dir = mock.Mock(return_value='')
-        with mock.patch('mesonbuild.interpreter.Interpreter._redetect_machines', mock.Mock()), \
+        with mock.patch('mesonbuild.interpreter.interpreter.Interpreter._redetect_machines', mock.Mock()), \
```

Comment:
Can you give more details about this module name clash?

Reply:
` AttributeError: <module 'mesonbuild.interpreter.Interpreter' from 'C:\\env\\meson\\mesonbuild\\interpreter\\Interpreter.py'> does not have the attribute '_redetect_machines'`

It tries to load `_redetect_machines` from `interpreter.py` module instead of `Interpreter` class. Not sure why it does like this on my machine and not in CI...

Reply:
```
<module 'mesonbuild.interpreter.Interpreter' from 'C:\\env\\meson\\mesonbuild\\interpreter\\Interpreter.py'>
```

This seems extremely wrong and will not be fixed by this change, probably.

- `mesonbuild.interpreter` is a package -- it has an `__init__.py` which imports `from .interpreter import Interpreter`
- `mesonbuild.interpreter.interpreter` is a module, interpreter.py, while `mesonbuild.interpreter.Interpreter` is a class object

What I don't understand is -- when python sees mesonbuild.interpreter.Interpreter, it should *first* load `mesonbuild/interpreter/__init__.py`, then evaluate the scope of such, and have mesonbuild.interpreter.Interpreter as a symbol manually defined at package scope. So it should not be trying to load it as a file instead.

Reply:
I investigated further. It seems the bug only occurs when meson is in editable install in my venv. With a normal install, or without the venv, the test passes. I still don't understand, but I guess I can drop this commit for now.